public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -44,7 +44,10 @@ import LanguageProvider from './language_provider';
 import { serializeParams } from '../../../core/utils/fetch';
 import { RowContextOptions } from '@grafana/ui/src/components/Logs/LogRowContextProvider';
 
-export type RangeQueryOptions = Pick<DataQueryRequest<LokiQuery>, 'range' | 'intervalMs' | 'maxDataPoints' | 'reverse'>;
+export type RangeQueryOptions = Pick<
+  DataQueryRequest<LokiQuery>,
+  'range' | 'intervalMs' | 'maxDataPoints' | 'reverse' | 'liveStreaming' | 'showingGraph' | 'showingTable'
```

Comment:
This RangeQueryOptions type feels very mess, is there an easier way to define it?

Reply:
You're right. I'll take a look and see if I can improve it